src/pages/travel/visa/index.tsx

```diff
@@ -206,78 +247,148 @@ const VisaPage: React.FC = () => {
-                className='h-[800px] overflow-y-auto pr-2'
-                role='listbox'
-                aria-label={t('countryList.ariaLabel')}
+      <div className='container mx-auto max-w-6xl py-4 md:py-12 px-4'>
+        {/* View Toggle and Search Bar */}
+        <div className='bg-white rounded-lg shadow-md p-4 mb-6'>
+          <div className='flex flex-col md:flex-row justify-between items-center gap-4'>
+            <div className='relative w-full md:w-96'>
+              <Search className='absolute left-3 top-3 h-5 w-5 text-gray-400' />
+              <input
+                ref={searchInputRef}
+                type='text'
+                placeholder={t('quickCheck.searchPlaceholder')}
+                className='w-full pl-10 pr-4 py-2 border border-gray-300 rounded-md focus:outline-hidden focus:ring-2 focus:ring-blue-500'
+                value={searchTerm}
+                onChange={e => setSearchTerm(e.target.value)}
+                aria-label={t('quickCheck.searchAriaLabel')}
+              />
+            </div>
+            <div className='flex items-center gap-2'>
+              <button
+                onClick={() => setViewMode('list')}
+                className={`flex items-center gap-2 px-4 py-2 rounded-md transition-colors ${
+                  viewMode === 'list'
+                    ? 'bg-blue-600 text-white'
+                    : 'bg-gray-100 text-gray-700 hover:bg-gray-200'
+                }`}
               >
-                {filteredCountries.length > 0 ? (
-                  filteredCountries.map(country => (
-                    <button
-                      key={country}
-                      className={`w-full text-left px-4 py-3 rounded-md mb-1 transition-colors ${
-                        selectedCountry === country
-                          ? 'bg-blue-100 text-blue-800 font-medium'
-                          : 'hover:bg-gray-100'
-                      }`}
-                      onClick={() => checkVisaRequirement(country)}
-                      role='option'
-                      aria-selected={selectedCountry === country}
-                    >
-                      {country}
-                    </button>
-                  ))
-                ) : (
-                  <div className='text-center py-8 text-gray-800'>
-                    <Search className='mx-auto h-8 w-8 mb-2 opacity-50' />
-                    <p>{t('countryList.noResults')}</p>
+                <Grid3x3 className='h-4 w-4' />
+                Grid View
+              </button>
+              <button
+                onClick={() => setViewMode('detail')}
+                className={`flex items-center gap-2 px-4 py-2 rounded-md transition-colors ${
+                  viewMode === 'detail'
+                    ? 'bg-blue-600 text-white'
+                    : 'bg-gray-100 text-gray-700 hover:bg-gray-200'
+                }`}
+              >
+                <List className='h-4 w-4' />
+                Detail View
```

Comment:
consider renaming to list view, i think detail view is confusing but this is subjective

Reply:
Oops totally missed this one, it was originally "list" then I changed it to "grid". This is an artifact of the change. Will fix.